cni-plugin/internal/pkg/utils/utils.go

```diff
@@ -363,6 +357,13 @@ func replaceHostLocalIPAMPodCIDR(logger *logrus.Entry, rawIpamData interface{},
 
 		ipamData["subnet"] = ipv6Cidr
 		logger.Infof("Calico CNI passing podCidrv6 to host-local IPAM: %s", ipv6Cidr)
+		return nil
+	}
+
+	// updateHostLocalIPAMDataForOS is only required for Windows and only ipv4 is supported
+	err := updateHostLocalIPAMDataForOS(subnet, ipamData)
+	if err != nil {
+		return err
```

Comment:
This makes sense now after looking at the version before the bug was introduced: https://github.com/projectcalico/calico/blob/79b442a53adb7d7f1fd62927d9322daf87dce9de/cni-plugin/internal/pkg/utils/utils.go#L329-L343